### PR TITLE
Add test for balance proof update after lock expiry

### PR DESCRIPTION
**Pull request by palango**
@hackaugusto The test for the suspected issue from this morning. Seems like your right.

Edit: I added a fix as well, but I'm not sure if this is the proper way to fix it.

### CODE REVIEW

**Comment by LefterisJP** [comment]
@palango can you also add a regression test for the partner (receiver) balance proof?

**Comment by palango** [comment]
@LefterisJP The test for the partner-side can be easily added to one of the tests augusto added https://github.com/raiden-network/raiden/pull/2834/files#diff-1338dac76122f31eb8ae5894300cc752R892

**Comment by LefterisJP** [comment]
@palango It's fine for me as long as one of you creates a regression test for that and ends up in master :)

Can I see a green build for this PR then to review and merge?

**Comment by LefterisJP** [comment]
I am looking at the failing test: `raiden/tests/integration/test_regression.py::test_regression_revealsecret_after_secret`

File: raiden/transfer/channel.py [hunk 1]
```diff
@@ -1431,20 +1431,22 @@ def events_for_close(
...
         pseudo_random_generator: random.Random,
-) -> typing.List[SendLockExpired]:
-    nonce = get_next_nonce(channel_state.our_state)
-    locked_amount = get_amount_locked(channel_state.our_state)
-    our_balance_proof = channel_state.our_state.balance_proof
+        chain_id: typing.ChainID,
+        token_network_identifier: typing.TokenNetworkID,
+        channel_identifier: typing.ChannelID,
+        recipient: typing.Address,
+):
```
Comment (ulope) [inline raiden/transfer/channel.py]: A return type annotation would be nice

Comment (LefterisJP) [reply raiden/transfer/channel.py]: It was added in: https://github.com/raiden-network/raiden/pull/2846/commits/d23666acf8ade6b67f3c4455e43b1546efe1d08b

File: raiden/tests/unit/test_channelstate.py [hunk 2]
```diff
@@ -1194,7 +1195,77 @@ def test_channel_must_never_expire_locks_with_onchain_secret():
...
+
+    is_valid, msg, _ = channel.is_valid_lock_expired(
+        state_change=lock_expired,
+        channel_state=channel_state,
+        sender_state=channel_state.partner_state,
+        receiver_state=channel_state.our_state,
+        block_number=block_number,
+    )
+
+    assert is_valid, msg
```
Comment (hackaugusto) [inline raiden/tests/unit/test_channelstate.py]: Hummmmm, this is really weird, the test is named `test_channel_must_never_expire_locks_with_onchain_secret` however this is asserting on the lock_expired *being accepted*.

It looks like this test and the one bellow got swapped.

**Pull request merged**